File: regex.js
const fs = require("node:fs");

const sstring = fs.readFileSync("sample.txt").toString();
// regex literals aren't necessary but are recommended for JavaScript regex
const pattern = RegExp(/^h.\D\S[a-z]{3,5}(...)\w+\d*$/gmd);
const pattern1 = RegExp("^h.\\D\\S[a-z]{3,5}(...)\\w+\\d*$", "gmd");
if (String(pattern) !== String(pattern1)) {
    console.log("Compiled regex patterns are unequal.");
}
[...sstring.matchAll(pattern)].forEach((result) => {
    console.log(result[0]);
    console.log(result[1]);
    console.log(result.indices[1][0]);
    console.log(result.indices[1][1]);
});

console.log(sstring.replace(/bob$/gm, "alice"));

console.log(/hello[0-9]+world/.test("hello42world"));
console.log(/^he[^A-Z]*\d{2}wo\w{2,}$/.test("hello42world"));
"Words, words, words.".split(/\W+/).forEach((matchString) => {
    console.log(matchString);
});

File: Regex.cs
using System.Text.RegularExpressions;

string sstring = File.ReadAllText("sample.txt");
string pattern = "(?m:^h.\\D\\S[a-z]{3,5}(...)\\w+\\d*$)";
foreach (Match match in Regex.Matches(sstring, pattern))
{
    Console.WriteLine(match.Groups[0].ToString());
    Console.WriteLine(match.Groups[1].ToString());
    Console.WriteLine(match.Groups[1].Index);
    Console.WriteLine(match.Groups[1].Index + match.Groups[1].Length);
    Console.WriteLine();
}

Console.WriteLine(MyRegex().Replace(sstring, "alice"));

Console.WriteLine(MyRegex1().IsMatch("hello42world"));
Console.WriteLine(MyRegex2().IsMatch("hello42world"));
string[] matchArray = MyRegex3().Split("Words, words, words.");
foreach (string match in matchArray)
{
    Console.WriteLine(match);
}

partial class Program
{
    [GeneratedRegex("(?m:bob$)")]
    private static partial Regex MyRegex();
}

partial class Program
{
    [GeneratedRegex("hello[0-9]+world")]
    private static partial Regex MyRegex1();
}

partial class Program
{
    [GeneratedRegex("^he[^A-Z]*\\d{2}wo\\w{2,}$")]
    private static partial Regex MyRegex2();
}

partial class Program
{
    [GeneratedRegex("\\W+")]
    private static partial Regex MyRegex3();
}
